siphon/simplewebservice/mesowest.py

```diff
@@ -0,0 +1,124 @@
+import warnings
+
+from bs4 import BeautifulSoup
+import numpy as np
+import pandas as pd
+
+from ..http_util import HTTPEndPoint
+
+warnings.simplefilter("ignore")
+
+class MesoWest(HTTPEndPoint):
+    """Download and parse data from the University of Utah's MesoWest archive."""
+    def __init__(self):
+        """Set up endpoint."""
+        super(MesoWest, self).__init__('https://mesowest.utah.edu/cgi-bin/droman/')
+
+    @classmethod
+    def request_data(cls, date, site_id):
+        r"""Retrieve upper air observations from the University of Utah MesoWest archive.
+
+        Parameters
+        ----------
+        date : datetime
+            The date of the desired observation.
+
+        site_id : str
+            The MesoWest identifier of the station for which data should be
+            downloaded.
+            https://mesowest.utah.edu/cgi-bin/droman/meso_station.cgi?area=1
+
+        kwargs
+            Arbitrary keyword arguments to use to initialize source
+
+        Returns
+        -------
+            :class:`pandas.DataFrame` containing the data
+
+        """
+        endpoint = cls()
+        df = endpoint._get_data(date, site_id)
+        return df
+
+    def _get_data(self, date, site_id):
+        """Download and parse upper air observations from an online archive.
+
+        Parameters
+        ----------
+        date : datetime
+            The date of the desired observation.
+
+        site_id : str
+            The MesoWest identifier of the station for which data should be
+            downloaded.
+            https://mesowest.utah.edu/cgi-bin/droman/meso_station.cgi?area=1
+
+        Returns
+        -------
+            :class:`pandas.DataFrame` containing the data
+
+        """
+        raw_data = self._get_data_raw(date, site_id)
+        soup = BeautifulSoup(raw_data, 'html.parser')
+
+        names_search = [[td.find_next(text=True).strip('\n\t\t') for td in tr.find_all('small')] for tr in soup.find_all('th')]
+        names = pd.DataFrame.from_records(names_search)[1].dropna(how='any', axis=0).reset_index(drop=True)
+
+        data_search = [[td.find_next(text=True).strip("\n\t\t") for td in tr.find_all("td")] for tr in soup.find_all('tr')]
+        df = pd.DataFrame.from_records(data_search).dropna(how='any', axis=0).reset_index(drop=True)
+        df = df.replace(r'^\s*$', np.nan, regex=True).replace('N/A', np.inf)
+
+        df[0] = pd.to_datetime(df[0]).apply(lambda x: x.time())
+
+        name = []
+        for i in range(1, len(df.columns)):
+            if str(df[i].dtypes) == 'object':
+                if str(df[i].iloc[0]).replace(' ', '').isalpha():
+                    pass
+                else:
+                    for j in range(0, len(df)):
+                        if '<' in str(df[i].iloc[j]).replace(' ', ''):
```

Comment:
If you're just checking for the presence of '>' and '<', you don't need to call `replace`. I think the whole block could be:

```python
if not ('<' in str(df[i].iloc[j]) or '>' in str(df[i].iloc[j])):
    df[i].iloc[j] = float(df[i].iloc[j])
```

Reply:
This seems like a much better method of handling the < and > symbols that sometimes occur in the visibility data. This will be implemented in the next commit.